src/main/java/hexgui/util/Options.java

```diff
@@ -0,0 +1,294 @@
+package hexgui.util;
+
+import java.io.*;
+import java.util.*;
+
+/** Parser for command line options. Options begin with a single '-' character. */
+public class Options {
+  /**
+   * Parse options.
+   *
+   * @param args Command line args from main method
+   * @param specs Specification of allowed options. Contains option names (without '-'). Options
+   *     that need an argument must have a ':' appended. The special argument '--' stops option
+   *     parsing, all following arguments are treated as non-option arguments.
+   * @throws Exception If options are not valid according to specs.
+   */
+  public Options(String[] args, String[] specs) throws Exception {
+    for (String spec : specs) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;specs&#39; (index: 1 | type: String[). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;args&#39;. The same parameter type is validated here for example: hexgui/src/main/java/hexgui/util/Platform.java:180. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).